rosbag2_cpp/src/rosbag2_cpp/info.cpp

```diff
@@ -52,4 +59,120 @@ rosbag2_storage::BagMetadata Info::read_metadata(
   return storage->get_metadata();
 }
 
+namespace
+{
+struct client_id_hash
+{
+  std::size_t operator()(const std::array<uint8_t, 16> & client_id) const
+  {
```

Comment:
```suggestion
struct client_id_hash
{
  static_assert(
    std::is_same<std::array<uint8_t, 16>,
    service_msgs::msg::ServiceEventInfo::_client_gid_type>::value);
  std::size_t operator()(const std::array<uint8_t, 16> & client_id) const
  {
```
Add static assert to verify assumption about `client_id` type.